### DISCUSSION TITLE
Are Nova Vue components considered "non-public API"?

### DISCUSSION THREAD

**Discussion by baszach** [laravel/nova-issues#4607] [Q&A] [answered]
Hello,
the previous developer of my Nova project used many Nova defined Vue components and mixins in our custom Nova fields, card etc. Some examples are `SelectControl.vue` or `HandlesActions.js`.

I would like to know if these components (or the frontend code in general) are considered non-public or public API by the maintainers. If they are considered non-public I will try not to use these components anymore since they could introduce breaking changes for me when updated.

Thanks in advance!

**Comment by crynobone** [comment, marked as answer]
Yes, they aren't public API. You should consider adding dusk tests for any component you override to ensure that everything is still working as expected.

**Reply by baszach** [reply]
Okay, great. Thanks for the reply.
